src/docker/local.rs

```diff
@@ -0,0 +1,93 @@
+use std::path::Path;
+use std::process::ExitStatus;
+
+use super::engine::*;
+use super::shared::*;
+use crate::cargo::CargoMetadata;
+use crate::errors::Result;
+use crate::extensions::CommandExt;
+use crate::{Config, Target};
+use atty::Stream;
+
+#[allow(clippy::too_many_arguments)] // TODO: refactor
+pub(crate) fn local_run(
```

Comment:
Rename `run` to `local_run`, so we can use `remote_run` later for if the command is remote.